Update Offering Page for PFIs (#1106)
TBD54566975/developer.tbd.website: site/docs/tbdex/pfi/creating-offerings.mdx

sidebar_position: 6
---

> # Creating Offerings

Reviewer: how come the name change?

Author: I think "creating offerings" is more narrow, this guide is actually covering offerings as a whole - this will also be more in line with upcoming changes I'd like to make explaining Exchanges

  </tr>
</table>

### Currencies

Both the `payinCurrency` and `payoutCurrency` fields are described by the [`CurrencyDetails`](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#currencydetails) data type, which has the following fields:

| Field        | Data Type | Description                                            |
|--------------|-----------|--------------------------------------------------------|
| `currencyCode` | string    | ISO 4217 currency code string. This field is required. |
| `minSubunits`  | string    | Minimum amount of currency that the offer is valid for |
| `maxSubunits`  | string    | Maximum amount of currency that the offer is valid for |

> To find more information on currency code strings to ensure your PFI uses a valid and correct value, see the [currency lists provided here](https://www.six-group.com/en/products-services/financial-information/data-standards.html).

Reviewer: whats this link? i dont see any currency lists

Author: The link actually provides 3 currency lists under "Current and Historical Lists" - I just can't link straight to the header. For whatever reason they divide it into three lists so there isn't just one to link to

### Payment Methods

Both the `payinMethods` and `payoutMethods` field are described by the [`PaymentMethod`](https://github.com/TBD54566975/tbdex/blob/main/specs/protocol/README.md#paymentmethod) data type, which has the following fields:

| Field                    | Data Type | Description                                                                                                                                                             |
|--------------------------|-----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `kind`                   | string    | Type of payment method (i.e. `DEBIT_CARD`, `CREDIT_CARD`, etc.). Note that because this field is of type string, the values are not standardized. This field is required. |
| `requiredPaymentDetails` | JSON      | A JSON schema containing the fields the Wallet must enter in order to use this payment method. This field is not enforced as required despite its name.                 |

> | `required`               | string[]  | A list of all the properties described `requiredPaymentDetails`.                                                                                                        |

Reviewer: I dont see the `required` and `feeSubunits` properties anymore in the API Guide
https://tbd54566975.github.io/tbdex-js/types/_tbdex_protocol.PaymentMethod.html

| `feeSubunits`            | string    | The fee expressed in the currency's sub units to make use of this payment method.                                                                                       |

When structuring your `requiredPaymentDetails` object, it will look similar to the following:

> ```json

Reviewer: This example is a bit abstract and makes it hard for me to imagine what the values should be. Should we maybe include an example with real values for the properties?

Author: Updated with concrete values

{
  "requiredPaymentDetails": {
    "$schema": "http://json-schema.org/draft-07/schema",
    "type": "object",
    "properties": {
      "property1": {
        "type": "string",
        "description": "A description of the first property",
        "property1MaxLength": 16,
      },
      "property2": {
        "type": "string",
        "description": "A description of the first property",
        "patternToMatch": "^(0[1-9]|1[0-2])\\/([0-9]{2})$"
      }
    }
  }
}
```

> Note that the while `type` and `description` fields are more intuitive and standard for you to include in your `requiredPaymentDetails` schema, you have the flexibility to detail any requirements you have in this object. Each `payinMethod` or `payoutMethod` can be represented by an array of `PaymentMethods`.

Reviewer: ```suggestion
Note that while `type` and `description` fields are more intuitive and standard for you to include in your `requiredPaymentDetails` schema, you have the flexibility to detail any requirements you have in this object. Each `payinMethod` or `payoutMethod` can be represented by an array of `PaymentMethods`.
```

> ### Credentials and Claims

Reviewer: ```suggestion
### Required Claims
```

> Before placing an Offer on an RFQ that a Wallet has submitted for your Offering, you'll likely want to verify some kind of information about the owner of that Wallet. To do this, you'll request a claim or claims from the Wallet in the form of [Verifiable Credentials (VCs)](/docs/web5/learn/verifiable-credentials). As a PFI, you only require specific knowledge of the VCs you need and how to request them; however if you'd like to learn more about [issuing](/docs/web5/build/verifiable-credentials/vc-issuance), [sharing](/docs/web5/build/verifiable-credentials/jwt-to-vc), and [storing](/docs/web5/build/verifiable-credentials/vcs-in-dwn) VCs, you can do so at each link. Additionally, you can learn about [building an end-to-end VC workflow here](/docs/web5/build/verifiable-credentials/fan-club-vc) if interested.

Reviewer: ```suggestion
In order to comply with certain financial regulations, you may require information about the customer you are transacting with. You can obtain this information via [Verifiable Credentials](/docs/web5/learn/verifiable-credentials). 
```

> You will specify the `requiredClaims` of your Offering by following [the DIF Presentation Definition](https://identity.foundation/presentation-exchange/#presentation-definition) - a way of defining what proof you as a Verifier require. You'll specify all required forms and details of proof, optional selection rules, and define flexibility where possible when different types of input may satisfy the requirements.

Reviewer: ```suggestion
To specify the exact credentials you need, you can do so in the `requiredClaims` section of your Offering by following [the DIF Presentation Definition](https://identity.foundation/presentation-exchange/#presentation-definition). You'll specify all required forms and details of proof, optional selection rules, and define flexibility where possible when different types of input may satisfy the requirements.
```

For a full breakdown of all the available fields and combinations, [see the DIF website](https://identity.foundation/presentation-exchange/#presentation-definition) for details. Below is a simple example of what a single required claim object could look like:

> ```json

Reviewer: use the Sanctions example as a more realistic use case for PFIs
https://developer.tbd.website/docs/tbdex/wallet/managing-credentials/#required-claims

Author: Added
